packages/core/versioning/core-versioned-http-server-internal/src/internal_versioned_router.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IRouter } from '@kbn/core-http-server';
+import type { VersionedRouter } from '@kbn/core-versioned-http-server';
+import type {
+  VersionedRoute,
+  VersionedRouteConfig,
+  VersionedRouterRoute,
+} from '@kbn/core-versioned-http-server';
+import { InternalVersionedRoute } from './internal_versioned_route';
+import { Method } from './types';
+
+export class InternalVersionedRouter implements VersionedRouter {
+  private readonly routes = new Set<InternalVersionedRoute>();
+  public static from({ router }: { router: IRouter }) {
+    return new InternalVersionedRouter(router);
+  }
+  private constructor(private readonly router: IRouter) {}
+
+  private registerVersionedRoute =
+    (routeMethod: Method) =>
+    (options: VersionedRouteConfig<Method>): VersionedRoute<Method, any> => {
+      const route = InternalVersionedRoute.from({
+        router: this.router,
+        method: routeMethod,
+        path: options.path,
+        options,
+      });
+      this.routes.add(route);
+      return route as VersionedRoute;
```

Comment:
NIT: do we need the `as VersionedRoute` casting here given the function's return type is already specified?

Reply:
Nope!